### PR TITLE
feat: add roadmap webpage

### PR DESCRIPTION
**Pull request by samajammin**
Add roadmap page (maci.pse.dev/roadmap) outlining the 2024 MACI team & product roadmap.

Fixes https://github.com/privacy-scaling-explorations/maci/issues/857

### CODE REVIEW

**Comment by samajammin** [comment]
Opening this up as a draft. I created this discussion to gather feedback from the community:
https://github.com/privacy-scaling-explorations/maci/discussions/859

Let's give that some time (a couple weeks?) before merging this. 

Alternatively, we could deploy this now (mentioning the roadmap is a proposal & link to the discussion) in order to get more eyes on that 🤔 - thoughts?

File: website/docusaurus.config.ts
Comment (samajammin) [inline website/docusaurus.config.ts:103]: Moved this "TypeDoc" nav link next to "Documentation" - I think it makes sense to keep those grouped together.

**Comment by samajammin** [comment]
> Alternatively, we could deploy this now (mentioning the roadmap is a proposal & link to the discussion) in order to get more eyes on that 🤔 - thoughts?

Decided to go this route, so we don't have to wait on merging this. Added this info banner:
Image 2023-12-04 at 14 16 48

Marking this PR as "Ready for review".

File: website/src/pages/roadmap.md
Comment (ctrlc03) [inline website/src/pages/roadmap.md:134]: these hackmd links seem now broken

File: website/src/pages/roadmap.md
Comment (ctrlc03) [inline website/src/pages/roadmap.md:264]: these hackmd links seem to be broken now

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Looking good, just a couple of broken links

**Review by baumstern — CHANGES_REQUESTED** [review]
It seems like integrating the team roadmap into MACI website might be out of its primary scope. Generally, project websites, such as [Solidity](https://soliditylang.org/) and [ethereum.org](https://ethereum.org/en/), do not include their roadmaps in the main navigation, suggesting it's not a common practice. I believe MACI site should focus on delivering topic-specific information, with the roadmap, being a meta aspect, perhaps being addressed similarly to how [ethereum.org](https://ethereum.org/en/) approaches it. They have linked their roadmap within their site, as seen here: https://ethereum.org/en/about/#roadmap, instead of directly hosting it. This approach could be more suitable for our site too.

**Comment by samajammin** [comment]
> It seems like integrating the team roadmap into MACI website might be out of its primary scope. Generally, project websites, such as [Solidity](https://soliditylang.org/) and [ethereum.org](https://ethereum.org/en/), do not include their roadmaps in the main navigation, suggesting it's not a common practice. I believe MACI site should focus on delivering topic-specific information, with the roadmap, being a meta aspect, perhaps being addressed similarly to how [ethereum.org](https://ethereum.org/en/) approaches it. They have linked their roadmap within their site, as seen here: https://ethereum.org/en/about/#roadmap, instead of directly hosting it. This approach could be more suitable for our site too.

So what are you suggesting @baumstern? To not have this page in the primary navigation?

Appreciate your feedback but I disagree. Perhaps in the longer term it shouldn't be on the site nav (e.g. when we create an "About" page, I'd be up for nesting the "Roadmap" page within there) but in short term, I think this makes sense. Given the MACI website only has a couple pages of content outside of our docs, I don't think ethereum.org or Solidity website are reasonable comparisons yet. This website is targeted for devs interested in integrating MACI. I suspect a top interest of theirs is "what are the plans for MACI?". People want to know what the vision & goals of the project are, which the Roadmap page addresses.

I do think your concern raises a valid question on "what should the sitemap of the website be?", so I'd be happy to create an issue to map that out.

**Comment by baumstern** [comment]
> I suspect a top interest of theirs is "what are the plans for MACI?". People want to know what the vision & goals of the project are, which the Roadmap page addresses.

Hmm I thought this is a project website (!= team website), such as a roadmap document should not be hosted directly on the project site. In that sense, ethereum.org and Solidity are still valid references.

However, it sounds like we want to create a accessible resource in a single place for now. If that's the case, then it's fine.

File: website/src/pages/roadmap.md
Comment (baumstern) [inline website/src/pages/roadmap.md:63]: Is https://maci.pse.dev/ intended to serve as a team website? If not, I suggest that sections pertaining to QF should ideally be hosted on https://github.com/privacy-scaling-explorations/qf/ instead

Comment (samajammin) [reply website/src/pages/roadmap.md:63]: > Is https://maci.pse.dev/ intended to serve as a team website

Yes, at least for now.

> I suggest that sections pertaining to QF should ideally be hosted on https://github.com/privacy-scaling-explorations/qf/ instead

I agree. I think eventually we should/will move things there. But given how small the team is & we don't have a solid website for QF, I'd like to keep things together for simplicity.

Comment (ctrlc03) [reply website/src/pages/roadmap.md:63]: @samajammin @baumstern is this conversation resolved?

**Comment by samajammin** [comment]
> Looking good, just a couple of broken links

Thanks for catching @ctrlc03. Should be fixed now - I replaced those with links to new GH issues.

**Comment by samajammin** [comment]
Hmmm @ctrlc03 do you know how I can dismiss @baumstern's change requests? Looks like I'm blocked on merging this but @baumstern is out for the rest of the year...

**Comment by ctrlc03** [comment]
> Hmmm @ctrlc03 do you know how I can dismiss @baumstern's change requests? Looks like I'm blocked on merging this but @baumstern is out for the rest of the year...

I believe you can uncheck "Do not allow bypassing the above settings" within the branch protection rule to then merge this and then can check that box back. Will need to solve conflicts on this one first though

**Review by baumstern — APPROVED** [review]
LGTM

**Comment by samajammin** [comment]
Sweet, thanks @baumstern @ctrlc03 🎉

**Pull request merged**
